idseq_pipeline/commands/non_host_alignment_functions.py

```diff
@@ -278,6 +291,80 @@ def log_corrupt(is_corrupt, m8_file, line):
         raise AssertionError
 
 
+def generate_taxon_count_json_from_m8(m8_file, hit_level_file, e_value_type, count_type, stats, output_file):
+    lineage_path = fetch_reference(LINEAGE_SHELF)
+    lineage_map = shelve.open(lineage_path)
+    taxid_properties = {}
+    hit_f = open(hit_level_file, 'rb')
+    m8_f = open(m8_file, 'rb')
+    # lines in m8_file and hit_level_file correspond (same read_id)
+    hit_line = hit_f.readline()
+    m8_line = m8_f.readline()
+    while hit_line and m8_line:
+        hit_line_columns = hit_line.rstrip("\n").split("\t")
+        read_id = hit_line_columns[0]
+        hit_level = hit_line_columns[1]
+        hit_taxid = hit_line_columns[2]
+        if int(hit_level) < 0:
+            hit_line = hit_f.readline()
+            m8_line = m8_f.readline()
+            continue
+        m8_line_columns = m8_line.split("\t")
+        assert m8_line_columns[0] == hit_line_columns[0], "read_ids in %s and %s do not match: %s vs. %s" % (os.path.basename(m8_file),
```

Comment:
This assert fired in https://us-west-2.console.aws.amazon.com/cloudwatch/home?region=us-west-2#logEventViewer:group=/aws/batch/job;stream=aegea_batch/default/9de9d30e-6050-4dcd-9e43-7c0f37b9b372

Reply:
oops: `JF317013.1-80 vs. JF317013.1-8`
Need to be more specific with my `grep` and include the trailing tab